packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -751,6 +752,34 @@ void main() {
       expect(desktop.values.union(mobile.values), equals(all.values));
     });
   });
+
+  group('timers pending after widget tree disposal', () {
+    testWidgets('no timers pending check', (WidgetTester tester) async {
+      final Timer timer = Timer(const Duration(seconds: 1), () {});
+      expect(timer.isActive, true);
+      timer.cancel();
+      expect(timer.isActive, false);
+    });
+
+    testWidgets('timers pending check', (WidgetTester tester) async {
+      expect(
+          () {
+            final FakeAsync timers = FakeAsync();
+            debugPrint('Pending timers:');
+            for (final FakeTimer timer in timers.pendingTimers) {
+              debugPrint(
+                  'Timer (duration: ${timer.duration}, '
+                      'periodic: ${timer.isPeriodic}), created:');
+              debugPrintStack(stackTrace: timer.creationStackTrace);
+              debugPrint('');
+            }
+            throw AssertionError('A Timer is still pending even after the widget tree was disposed.');
```

Comment:
This assertion should be the coming from binding.dart#1132.  Have you tried to see what happens if you add a nested `testWidgets`, for example:

```dart
expect(() {
  testWidgets('Code under test', (WidgetTester tester) async {
      // Add timer
  });
}, throwsA(...));
```

Reply:
I tried that, but I get an error saying "expect() may only be called within a test." I also tried nesting the expect() within another test, but that didn't work either.

I added a timer in the expect, but the closure returns null. I've also tried checking if tester.takeException() is not null after calling timer.

Reply:
It turns out you could do something like this:

```dart
group('Pending timer', () {
  TestExceptionReporter currentTestException;

  setUp(() {
    currentTestException = reportTestException;
  });

  tearDown(() {
    reportTestException = currentTestException;
  });

  test('Throws assertion message without code', () async {
    FlutterErrorDetails flutterErrorDetails;
    reportTestException = (FlutterErrorDetails details, String testDescription) {
      flutterErrorDetails = details;
    };

    final TestWidgetsFlutterBinding binding = TestWidgetsFlutterBinding.ensureInitialized() as TestWidgetsFlutterBinding;
    await binding.runTest(() async {
      final Timer timer = Timer(const Duration(seconds: 1), () {});
      expect(timer.isActive, true);
    }, () {});

    expect(flutterErrorDetails, isNotNull);
    expect(flutterErrorDetails.exception, isA<AssertionError>());
    expect(flutterErrorDetails.exception.message, 'A Timer is still pending even after the widget tree was disposed.');
  });
});
```

@Hixie wrote the package. I'm not sure if there's a simpler way of testing this behavior for #53296.